src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2024,34 +2081,40 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
 
   g_autofree char *version_key = g_strdup_printf ("%d", n_deployments - ostree_deployment_get_index (deployment));
   ostree_bootconfig_parser_set (bootconfig, OSTREE_COMMIT_META_KEY_VERSION, version_key);
-  g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", kernel_layout->kernel_namever, NULL);
-  ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
+
+  if (!payg)
+    {
+      g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", kernel_layout->kernel_namever, NULL);
+      ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
+    }
 
   val = ostree_bootconfig_parser_get (bootconfig, "options");
   g_autoptr(OstreeKernelArgs) kargs = ostree_kernel_args_from_string (val);
 
-  if (kernel_layout->initramfs_namever)
+  if (!payg)
     {
-      g_autofree char * initrd_boot_relpath =
-        g_strconcat ("/", bootcsumdir, "/", kernel_layout->initramfs_namever, NULL);
-      ostree_bootconfig_parser_set (bootconfig, "initrd", initrd_boot_relpath);
+      if (kernel_layout->initramfs_namever)
+        {
+          g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", kernel_layout->initramfs_namever, NULL);
+          ostree_bootconfig_parser_set (bootconfig, "initrd", boot_relpath);
 
-      if (overlay_initrds)
+          if (overlay_initrds)
+            {
+              g_ptr_array_add (overlay_initrds, NULL);
+              ostree_bootconfig_parser_set_overlay_initrds (bootconfig, (char**)overlay_initrds->pdata);
+            }
+        }
+      else
         {
-          g_ptr_array_add (overlay_initrds, NULL);
-          ostree_bootconfig_parser_set_overlay_initrds (bootconfig, (char**)overlay_initrds->pdata);
+          g_autofree char *prepare_root_arg = NULL;
+          prepare_root_arg = g_strdup_printf ("init=/ostree/boot.%d/%s/%s/%d/usr/lib/ostree/ostree-prepare-root",
+                                                 new_bootversion, osname, bootcsum,
+                                                 ostree_deployment_get_bootserial (deployment));
+          ostree_kernel_args_replace_take (kargs, g_steal_pointer (&prepare_root_arg));
         }
     }
-  else
-    {
-      g_autofree char *prepare_root_arg = NULL;
-      prepare_root_arg = g_strdup_printf ("init=/ostree/boot.%d/%s/%s/%d/usr/lib/ostree/ostree-prepare-root",
-                                             new_bootversion, osname, bootcsum,
-                                             ostree_deployment_get_bootserial (deployment));
-      ostree_kernel_args_replace_take (kargs, g_steal_pointer (&prepare_root_arg));
-    }
 
-  if (kernel_layout->devicetree_namever)
+  if (kernel_layout->devicetree_namever && !payg)
```

Comment:
theoretically, I think this entire if () {} else { } should be avoided when !payg. Either by adding an  && !payg on line 2122 or nesting the whole mess here. (in practice I don't think we'll see device tree + payg anyway...)

Reply:
Thanks for reviewing. I wasn't focusing that much on the surrounding logic. I'll try to clean that up.

Reply:
I thought about nesting more things under a large `if (!payg)` block but ultimately decided to sprinkle in some more `&& !payg`s around so it's less invasive. See 24ae817a15f74edd1177b14bca2d8d520b0ca3cd.

Reply:
I like what you've done - I thought I wanted it the other way around, but seeing it like this it's clear that not re-indenting half the file is probably much easier to manage for future rebases! :)

Reply:
I think you could go a little further and change some `else` to `else if (!payg)` to leave some of the indentation alone, but I think I've monkeyed with this enough for now.